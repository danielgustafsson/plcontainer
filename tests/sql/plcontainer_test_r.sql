select rlog100();
select rbool('t');
select rbool('f');
select rint(NULL::int2);
select rint(123::int2);
select rint(234::int4);
select rint(345::int8);
select rfloat(3.1415926535897932384626433832::float4);
select rfloat(3.1415926535897932384626433832::float8);
select rtext('123');
select rtest_mia();
select vec8('{1.23, 1.32}'::float8[]);
select vec8('{1, 5,10, 100, 7}'::int8[]);
select vec4('{1.23, 1.32}'::float4[]);
select vec4('{1, 5,10, 100, 7}'::int4[]);
select rlog100_shared();
select rpg_spi_exec('select 1');
--select paster('{hello, happy}','{world, birthday}',' ');
--select rtest_spi_tup('select fname, lname,username from users order by 1,2,3');
-- This function is of "return setof record" type which is not supported yet
-- select rtest_spi_ta('select oid, typname from pg_type where typname = ''oid'' or typname = ''text''');
