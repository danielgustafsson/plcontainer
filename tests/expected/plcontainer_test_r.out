set datestyle='ISO,MDY';
select rlog100();
 rlog100 
---------
 2
(1 row)

select rbool('t');
 rbool 
-------
 t
(1 row)

select rbool('f');
 rbool 
-------
 f
(1 row)

select rint(NULL::int2);
 rint 
------
    0
(1 row)

select rint(123::int2);
 rint 
------
  124
(1 row)

select rint(234::int4);
 rint 
------
  236
(1 row)

select rint(345::int8);
 rint 
------
  348
(1 row)

select rfloat(3.1415926535897932384626433832::float4);
 rfloat  
---------
 4.14159
(1 row)

select rfloat(3.1415926535897932384626433832::float8);
      rfloat      
------------------
 5.14159265358979
(1 row)

select rnumeric(3.1415926535897932384626433832::numeric);
     rnumeric     
------------------
 6.14159265358979
(1 row)

select rtimestamp('2012-01-02 12:34:56.789012'::timestamp);
         rtimestamp         
----------------------------
 2012-01-02 13:34:56.789011
(1 row)

select rtimestamptz('2012-01-02 12:34:56.789012 UTC+4'::timestamptz);
         rtimestamptz          
-------------------------------
 2012-01-02 08:34:56.789012-08
(1 row)

select rtext('123');
 rtext  
--------
 123foo
(1 row)

select rtest_mia();
         rtest_mia          
----------------------------
 {{1,2,3,4,5},{6,7,8,9,10}}
(1 row)

select vec('{1.23, 1.32}'::float8[]);
     vec     
-------------
 {1.23,1.32}
(1 row)

select vec('{1, 5,10, 100, 7}'::int8[]);
      vec       
----------------
 {1,5,10,100,7}
(1 row)

select vec('{1.23, 1.32}'::float4[]);
     vec     
-------------
 {1.23,1.32}
(1 row)

select vec('{1, 5,10, 100, 7}'::int4[]);
      vec       
----------------
 {1,5,10,100,7}
(1 row)

select rintarr('{1,2,3,4}'::int2[]);
 rintarr 
---------
      10
(1 row)

select rintarr('{1,2,3,null}'::int2[]);
 rintarr 
---------
       6
(1 row)

select rintarr('{null}'::int2[]);
 rintarr 
---------
       0
(1 row)

select rintarr('{1,2,3,4,5}'::int4[]);
 rintarr 
---------
      15
(1 row)

select rintarr('{1,2,3,4,null}'::int4[]);
 rintarr 
---------
      10
(1 row)

select rintarr('{null}'::int4[]);
 rintarr 
---------
       0
(1 row)

select rintarr('{1,2,3,4,6}'::int8[]);
 rintarr 
---------
      16
(1 row)

select rintarr('{1,2,3,6,null}'::int8[]);
 rintarr 
---------
      12
(1 row)

select rintarr('{null}'::int8[]);
 rintarr 
---------
       0
(1 row)

select rfloatarr('{1.2,2.3,3.4,5.6}'::float8[]);
 rfloatarr 
-----------
      12.5
(1 row)

select rfloatarr('{1.2,2.3,3.4,null}'::float8[]);
 rfloatarr 
-----------
       6.9
(1 row)

select rfloatarr('{null}'::float8[]);
 rfloatarr 
-----------
         0
(1 row)

select rfloatarr('{1.2,2.3,3.4,5.6,6.7}'::float4[]);
 rfloatarr 
-----------
      19.2
(1 row)

select rfloatarr('{1.2,2.3,3.4,5.6,null}'::float4[]);
 rfloatarr 
-----------
      12.5
(1 row)

select rfloatarr('{null}'::float4[]);
 rfloatarr 
-----------
         0
(1 row)

select rboolarr('{1,1,0}'::bool[]);
 rboolarr 
----------
        2
(1 row)

select rboolarr('{1,1,0,NULL}'::bool[]);
 rboolarr 
----------
        2
(1 row)

select rboolarr('{NULL}'::bool[]);
 rboolarr 
----------
        0
(1 row)

select rlog100_shared();
 rlog100_shared 
----------------
 2
(1 row)

select rpg_spi_exec('select 1');
 rpg_spi_exec 
--------------
 1
(1 row)

--select paster('{hello, happy}','{world, birthday}',' ');
--select rtest_spi_tup('select fname, lname,username from users order by 1,2,3');
-- This function is of "return setof record" type which is not supported yet
-- select rtest_spi_ta('select oid, typname from pg_type where typname = ''oid'' or typname = ''text''');
